### PR TITLE
Add replicated log storage provider

### PR DESCRIPTION
**Pull request by aquamatthias**
- make the store_backend parameter available (no hidden internal_store_backend)
- added parameter for replicated log parameters

### CODE REVIEW

File: src/test/scala/mesosphere/util/state/mesos/MesosStateZKStoreTest.scala [hunk 1]
```diff
@@ -9,7 +9,7 @@ import org.scalatest.{ ConfigMap, Matchers }
 
 import scala.concurrent.duration._
 
-class MesosStateStoreTest extends PersistentStoreTest with StartedZookeeper with Matchers {
```
Comment (kolloch) [inline src/test/scala/mesosphere/util/state/mesos/MesosStateZKStoreTest.scala]: Was that on purpose?

Comment (kolloch) [inline src/test/scala/mesosphere/util/state/mesos/MesosStateZKStoreTest.scala]: It was, got it know.

File: src/main/scala/mesosphere/marathon/MarathonConf.scala [hunk 2]
```diff
@@ -161,14 +161,34 @@ trait MarathonConf extends ScallopConf with ZookeeperConf with IterativeOfferMat
     noshort = true
   )
 
-  //Internal settings, that are not intended for external use
-  lazy val internalStoreBackend = opt[String]("internal_store_backend",
-    descr = "The backend storage system to use. One of zk, mesos_zk, mem",
+  lazy val storeBackend = opt[String]("store_backend",
```
Comment (kolloch) [inline src/main/scala/mesosphere/marathon/MarathonConf.scala]: We talked about this: Maybe we want to avoid renaming this parameter.

**Comment by kolloch** [comment]
Don't merge. This is in the process of being tested.

File: src/main/scala/mesosphere/marathon/MarathonConf.scala (hunk 2 above)
Comment (aquamatthias) [inline src/main/scala/mesosphere/marathon/MarathonConf.scala]: Nobody should know that parameter. Do you think this is a problem?
I would like to reserve `--internal*` for experimental flags.

**Comment by aquamatthias** [comment]
Rebase on master - close this one.
@kolloch thanks for feedback

**Pull request closed without merging**